### PR TITLE
Add generated dates (#864)

### PR DESCRIPTION
**Pull request by beckjake**
Add a field named `generated_at` to the root of the catalog and the manifest, which contains an RFC3339-formatted timestamp.

### CODE REVIEW

File: dbt/contracts/graph/parsed.py [hunk 1]
```diff
@@ -252,6 +252,9 @@
     'properties': {
         'nodes': PARSED_NODES_CONTRACT,
         'macros': PARSED_MACROS_CONTRACT,
+        'generated_at': {
+            'type': 'date-time'
```
Comment (cmcarthur) [inline dbt/contracts/graph/parsed.py]: the correct json schema draft4 type for this is:

```json
{
    "type": "string",
    "format": "date-time"
}
```

File: test/integration/029_docs_generate_tests/test_docs_generate.py [hunk 2]
```diff
@@ -43,6 +44,25 @@ def run_and_generate(self, extra=None):
...
+        yesterday = now + timedelta(days=-1)
+        parsed = datetime.strptime(timestr, '%Y-%m-%dT%H:%M:%S.%fZ')
+        self.assertLess(
+            yesterday, parsed,
+            'parsed date {} happened over 24h ago'.format(parsed)
+        )
+        self.assertGreater(
+            now, parsed,
+            'parsed date {} happened in the future'.format(parsed)
+        )
```
Comment (cmcarthur) [inline test/integration/029_docs_generate_tests/test_docs_generate.py]: hah interesting note about freezegun.freeze_time. i think this is fine as a single-use hack. let's make sure not to overuse this.

File: dbt/contracts/graph/parsed.py (hunk 1 above)
Comment (beckjake) [reply dbt/contracts/graph/parsed.py]: So, I was wondering how tests were passing after this comment - it turns out we never use this schema at all. Should I just remove it entirely?

Comment (cmcarthur) [reply dbt/contracts/graph/parsed.py]: we need at least one json schema for the manifest, and at least one for the catalog. it's actually really important as a piece of documentation to share with the community. so ideally we'd either (a) apply it or (b) just leave it in here but do our best to correct it, even though it's not currently used. up to you which you want to do here

Comment (beckjake) [reply dbt/contracts/graph/parsed.py]: I've changed the ParsedManifest to validate the schema (and made sure it fails on the old schema).

**Pull request merged**